utils/crypto/bls/signer/localsigner/bls_test.go

```diff
@@ -141,20 +151,24 @@ func TestIncorrectMessageAggregation(t *testing.T) {
 
 func TestDifferentMessageAggregation(t *testing.T) {
 	require := require.New(t)
-	sk1, pk1 := NewKeyPair(require)
-	sk2, pk2 := NewKeyPair(require)
-	sk3, pk3 := NewKeyPair(require)
-
-	pks := []*bls.PublicKey{pk1, pk2, pk3}
+	signers := []*LocalSigner{NewSigner(require), NewSigner(require), NewSigner(require)}
+	pks := Map(signers, func(signer *LocalSigner) *bls.PublicKey {
+		return signer.PublicKey()
+	})
 
 	msg := []byte("TestDifferentMessagesAggregation local signer")
 	differentMsg := []byte("TestDifferentMessagesAggregation local signer with different message")
 
-	signatures := []*bls.Signature{
-		sk1.Sign(msg),
-		sk2.Sign(msg),
-		sk3.Sign(differentMsg),
-	}
+	diffSig, err := signers[0].Sign(differentMsg)
+	require.NoError(err)
+
+	sameSigs := Map(signers[1:], func(signer *LocalSigner) *bls.Signature {
+		sig, err := signer.Sign(msg)
+		require.NoError(err)
+		return sig
+	})
```

Comment:
I personally find:
```suggestion
	sameSigs := make([]*bls.Signature, len(signers)-1)
	for i, signer := range signers[1:] {
		sig, err := signer.Sign(msg)
		require.NoError(err)
		sameSigs[i] = sig
	}
```
more natural for a golang dev. Using stdlib/language features should generally be preferred over (unless there is an actual gain out of said implementation).

Basically I think the boilerplate of:
```go
newSlice := make([]T, len(oldSlice))
for i, v := range oldSlice {
    // XXX
   newSlice[i] = newV
}
```

Should be preferred over:
```go
newSlice := Map(oldSlice, func(T) V {
    // XXX
   return newV
})
```

cc: @ARR4N would like to hear your thoughts here.

Reply:
All that being said, I think the `(*testing.T, []byte, []*LocalSigner) -> []*bls.Signature` should be implemented as a helper function because it is used in a bunch of places.

Reply:
TL;DR I could be convinced of both arguments.

I'll think "out loud":

1. My ultimate goal in readability / style is to reduce cognitive load for the _reader_ of the code. This allows them (us) to more easily parse what's going on when trying to find bugs, make changes, etc.
2. Characteristics that feed into reduced cognitive load are idioms, abstractions, reduced nesting, blah blah blah. This could usually go unsaid, but I think it's important to call out idioms as "does this feel like Go" because if it doesn't then it's a distraction and that adds cognitive load.
3. For me, decisions like this boil down to cost-benefit analysis. For helper functions, the typical benefit is reducing code while the typical cost is distracting the reader by causing a cognitive branch as they read the helper. We don't have deep meat stacks in our skulls so that push/pop is expensive.

_Usually_ I'd say the saving here (the `make()`) isn't worth the distraction and that a little repetition isn't the end of the world (i.e. with @StephenButtolph). However `Map()`, `Reduce(), `Filter()` are all extremely well-known concepts so I don't think it's a distraction. If anything, it's a signal of intent to the reader, which is akin to the benefit of an idiom (i.e. with @richardpringle).

That was a very **long-winded way of saying that I'm on the fence** and quite stable but if I were to fall it would be towards a shared package that introduces those functions. Just please give it a name that communicates something (maybe `fn` because these are functional concepts) and not "underscore" like in JS! It's also not appropriate to export `bls.Map()`.

Reply:
I missed that this was just in tests as I only read Stephen's comment. If this were my PR I'd keep the function but rename it to something like `mapped()`. `Map()` being exported is very suggestive of it being a function in the package, not a test helper, which is a bit jarring. As I said before, I really don't have strong opinions either way (but do want a green bike shed).